doc/loader.rst

```diff
@@ -0,0 +1,81 @@
+==========
+MIG LOADER
+==========
+
+.. sectnum::
+.. contents:: Table of contents
+
+**Note**: mig-loader is a work in progress component and is not complete. This
+document is intended to summarize the concept and planned approach, and will be
+updated once additional progress is made.
+
+Overview
+--------
+In normal scenarios, the MIG agent is deployed using some form of package, for
+example RPM or deb. When the agent needs to be updated, the older agent package
+is removed from the system and a new package is installed.
+
+This approach works well for systems under some form of configuration
+management (e.g., services) as it is easy to deploy newer package versions
+to these devices. This approach does not work well with devices such as user
+workstations, as typically they are not managed in the same way.
+
+`mig-loader` is intended to act as bootstrapping function for the agent. The
+idea is, rather than install an agent on a system, `mig-loader` will be
+installed instead, and will manage the agent software on an on-going basis.
+
+Operation
+---------
+As an example, the loader would operate as follows.
+
+The loader binary would be installed on a client system using a package in the
+same way the agent would be installed now. This package would also be
+responsible for adding an operation to the scheduling facility on the system
+(for example, cron, Windows Scheduler) to run `mig-loader` on a periodic basis
+on the system.
+
+When the loader is run by the scheduler, it will create a manifest based on
+what MIG components/configuration files are present on the system. It is likely
+this will be limited to the agent itself, and configuration files used by the
```

Comment:
is this use to detect the version of mig-agent currently installed on the system and report that back to the api?